esmvaltool/diag_scripts/land_carbon_cycle/diag_global_turnover.py

```diff
@@ -0,0 +1,798 @@
+    # figure configuration
+    fig_config = _get_fig_config(diag_config)
+
+    # colormap configuration
+    cb_info = _get_diagonal_colorbar_info()
+
+    # define the figure and axis
+    plt.figure(figsize=(5, 3))
+    _ax = plt.axes([0.1, 0.1, 0.9, 0.9],
+                   projection=ccrs.Robinson(central_longitude=0),
+                   frameon=False)
+    # plot data over the map
+    plt.imshow(_get_data_to_plot(_dat.data),
+               norm=mpl.colors.BoundaryNorm(cb_info['tickBounds'],
+                                            len(cb_info['tickBounds'])),
+               cmap=cb_info['colMap'],
+               origin='upper',
+               vmin=cb_info['tickBounds'][0],
+               vmax=cb_info['tickBounds'][-1],
+               transform=ccrs.PlateCarree())
+    _fix_map(_ax)
+
+    # get the data and set the title of the map
+
+    _dat_median = np.nanmedian(
+        _remove_invalid(_dat.data, fill_value=fig_config['fill_value']))
+    title_str = (f'{_dat.long_name} ({_dat.units}), {_name},\n'
+                 f'global = {_datglobal:.2f}, median = {_dat_median:.2f}')
+
+    plt.title(title_str, fontsize=0.98 * fig_config['ax_fs'])
+
+    # draw the colorbar
+    _axcol_dia = [0.254, fig_config['y_colo_single'], 0.6, 0.035]
+    plut.mk_colo_tau(_axcol_dia,
+                     cb_info['tickBounds'],
+                     cb_info['colMap'],
+                     tick_locs=cb_info['ticksLoc'],
+                     cbfs=0.86 * fig_config['ax_fs'],
+                     cbtitle='',
+                     cbrt=90)
+
+    # save and close figure
+    t_x = plt.figtext(0.5, 0.5, ' ', transform=plt.gca().transAxes)
+    plut.save_figure(plot_path, _extr_art=[t_x])
+    plt.close()
+
+
+def main(diag_config):
+    """
+    Evaluate global distribution of ecosystem carbon turnover time.
+
+    Argument:
+    --------
+        diag_config - nested dictionary of metadata
+    """
+    my_files_dict = group_metadata(diag_config['input_data'].values(),
+                                   'dataset')
+
+    # get the data from the observation
+    global_tau_obs = _get_obs_data(diag_config)
+    base_name = ('{title}_{source_label}_'
+                 '{grid_label}'.format(
+                     title=global_tau_obs['grid']['tau_ctotal'].long_name,
+                     source_label=diag_config['obs_info']['source_label'],
+                     grid_label=diag_config['obs_info']['grid_label']))
+
+    global_tau_mod = {}
+    global_tau_mod['grid'] = {}
+    global_tau_mod['global'] = {}
+    provenance_record = get_provenance_record(
+        "Comparison of global distributions of turnover time of carbon"
+        "with matrix comparison, maps, and multimodel bias. Reproduces"
+        "figure 3 in Carvalhais et al. (2014).", ['mean', 'perc'], ['global'],
+        _get_ancestor_files(diag_config, 'tau_ctotal'))
+
+    with ProvenanceLogger(diag_config) as provenance_logger:
+        for key, value in my_files_dict.items():
```

Comment:
It would be a bit easier to understand this code if you used names like `dataset, filename` instead of `key, value` for your variables

Reply:
I will change these, and wherever applicable.

Reply:
changed accordingly in all diagnostic files.